src/main/java/com/solvd/pages/HomePage.java

```diff
@@ -1,51 +1,85 @@
 package com.solvd.pages;
 
+import com.solvd.Main;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.PageFactory;
 
-public class HomePage {
-    WebDriver driver;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.List;
 
+public class HomePage extends AbstractPage {
     @FindBy(css = "a.menu_home")
     WebElement homeButton;
 
     @FindBy(id = "main_menu")
     WebElement homeMenu;
 
+    @FindBy(xpath = "//a[text()='Login or register']")
+    WebElement loginButton;
+
+
     @FindBy(id = "filter_keyword")
     WebElement searchInput;
 
     @FindBy(className = "button-in-search")
     WebElement searchButton;
 
-    public HomePage(WebDriver driver) {
+    @FindBy(className = "block_7")
+    WebElement cartButton;
+
+    @FindBy(className = "productcart")
+    List<WebElement> productNames;
 
-        this.driver = driver;
-        PageFactory.initElements(driver, this);
+    private String url;
+
+    public HomePage(WebDriver driver) {
+        super(driver);
+        Properties properties = new Properties();
```

Comment:
loading properties is not the page's responsibility.
 Extract property loading to a utility class